updated ClipboardItem and Clipboard documentation and examples using … (#33262)
mdn/content: files/en-us/web/api/clipboarditem/index.md

- {{domxref("ClipboardItem.presentationStyle", "presentationStyle")}} {{ReadOnlyInline}}
  - : Returns one of the following: `"unspecified"`, `"inline"` or `"attachment"`.

## Static methods

_This interface defines the following methods._

> - {{domxref("ClipboardItem.supports", "supports()")}}

Reviewer: ```suggestion
- {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} 
```

>   - : Returns true or false whether a {{Glossary("MIME type")}} type is supported. This allows us to detect if a {{Glossary("MIME type")}} is supported by the clipboard before attempting to write data.

Reviewer 2: Nit: _before attempting to generate the data for the {{Glossary("MIME type")}} and write to the clipboard_.

Reviewer: ```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.
```

(although I would personally omit the second sentence and have it in the main page)

>   - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.

Reviewer 3: ```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows to detect if a MIME type is supported by the clipboard before attempting to write data.
```

The usage of "us" felt a bit inconsistent with other reference docs.

Reviewer 3: I see this page uses "us"/"we" in the examples section. I think it's fine there because it's more of a tutorial. But in the reference part, it feels a bit out of place.

Reviewer: I agree about "us" but "allows to detect" is not quite right. "enables to website to detect" is better if a bit longwinded.

### Writing to the clipboard

> Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.

Reviewer: ```suggestion
Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}}  to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```

> Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.

Reviewer: ```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'img/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

(however, if we decide not to use `supports()` with "img/png", per https://github.com/mdn/content/pull/33262#discussion_r1585203697, then obviously this won't apply, and we should just revert this part of the PR.)

Reviewer 4: Ah, this the same problem I see with https://github.com/mdn/content/pull/33262/files#r1597871059 ?  I guess it is OK to for one example to use a mandatory type, but we should have at least one example that does not and ideally cross link.

Reviewer 4: WillB suggestion updated with `image/` rather than `img`, and changed to new type.
I've decided that in this particular case we'll add a note after to indicate that this check isn't strictly necessary.

```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'image/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

### Writing to the clipboard

> Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`. Then we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if the MIME data type is supported before writing it to the clipboard.

Reviewer 2: Looks like we changed it to first check for the support of the PNG image MIME type and _then_ generate the data and write to the clipboard, right? Can we change this text to reflect the changes we made here?
